common/djangoapps/util/json_request.py

```diff
@@ -54,7 +56,14 @@ def parse_json_into_request(request, *args, **kwargs):
 
         return view_function(request, *args, **kwargs)
 
-    return parse_json_into_request
+    if isinstance(view_function, type) and issubclass(view_function, View):
+        view_function.dispatch = method_decorator(expect_json)(view_function.dispatch)
+        return view_function
+    else:
+        @wraps(view_function)
+        def wrapper(request, *args, **kwargs):
+            return parse_json_into_request(request, *args, **kwargs)
+        return wrapper
```

Comment:
I guess this is needed to make this decorator work with `APIView`?

Reply:
Yup that's right.